src/stepFunctions/commands/downloadStateMachineDefinition.ts

```diff
@@ -31,15 +32,25 @@ export async function downloadStateMachineDefinition(params: {
             params.stateMachineNode.details.stateMachineArn
         )
 
-        const wsPath = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders[0].uri.fsPath : '/'
-        let filePath = path.join(wsPath, params.stateMachineNode.details.name + '.asl.json')
-        const fileInfo = await vscode.window.showSaveDialog({ defaultUri: vscode.Uri.file(filePath) })
-        if (fileInfo) {
-            filePath = fileInfo.fsPath
-            fs.writeFileSync(filePath, stateMachineDetails.definition, 'utf8')
-            const openPath = vscode.Uri.file(filePath)
-            const doc = await vscode.workspace.openTextDocument(openPath)
-            vscode.window.showTextDocument(doc)
+        if (params.isDownloadAndRender) {
+            const doc = await vscode.workspace.openTextDocument({
+                language: 'asl',
+                content: stateMachineDetails.definition,
+            })
+
+            const textEditor = await vscode.window.showTextDocument(doc)
+            await vscode.commands.executeCommand('aws.previewStateMachine', textEditor)
+        } else {
+            const wsPath = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders[0].uri.fsPath : '/'
+            let filePath = path.join(wsPath, params.stateMachineNode.details.name + '.asl.json')
+            const fileInfo = await vscode.window.showSaveDialog({ defaultUri: vscode.Uri.file(filePath) })
+            if (fileInfo) {
+                filePath = fileInfo.fsPath
+                fs.writeFileSync(filePath, stateMachineDetails.definition, 'utf8')
+                const openPath = vscode.Uri.file(filePath)
```

Comment:
Is this equivalent to `fileInfo`?

Reply:
ok, this is preexisting code but I'm removing it. Yes, it is the same thing.

Reply:
that's a bit tricky. also tricky that `filePath` is reassigned. I would rename the partial path variables to make it more obvious, e.g. `relativePath` and `relativePathUri`. Or force things to be absolute at creation-time, then there is no confusion.

Reply:
Actually, I misunderstood the code because of (kind of bug) in it.
The first path is a default path that will be pre-selected for the user. Before it was defaulting to root directory when no folders were open which made me thing it's a relative path (I missed `/` at the beginning). 

The second path is the path the user actually selected (it can be different than default path).

I renamed variables to reflect that. Also, I added home directory as a fallback when no folders are selected in vscode.